grid/app/main/events/fl_events.py

```diff
@@ -92,6 +87,28 @@ def authenticate(message: dict, socket) -> str:
         response[RESPONSE_MSG.ERROR] = str(e)
 
     response = {MSG_FIELD.TYPE: FL_EVENTS.AUTHENTICATE, MSG_FIELD.DATA: response}
+    return response
+
+
+def authenticate(message: dict, socket) -> str:
+    """ New workers should receive a unique worker ID after authenticate on PyGrid platform.
+        Args:
+            message : Message body sended by some client.
+            socket: Socket descriptor.
+        Returns:
+            response : String response to the client
+    """
+    response = {}
+    _auth_token = message.get("auth_token")
+    model_name = message.get("model_name", None)
+
+    verification_result = verify_token(_auth_token, model_name)
+
+    if verification_result["status"] == RESPONSE_MSG.SUCCESS:
+        response = assign_worker({"auth_token": _auth_token}, None)
```

Comment:
Woops, I think we have big problem here:
1. Worker sends auth request 
2. We validate token against specific model config
3. We create worker_id and store in DB/workers table
4. Now worker with this worker_id is considered authenticated for any model because when requesting a cycle we don't check anything except worker_id presence in DB!
https://github.com/OpenMined/PyGrid/blob/3fbdebd49ce6a24b3d0ef6731942996b621cda67/grid/app/main/events/fl_events.py#L135-L147